src/app_service/service/community/service.nim

```diff
@@ -1659,7 +1717,13 @@ QtObject:
 
   proc isCommunityRequestPending*(self: Service, communityId: string): bool {.slot.} =
     for communityRequest in self.myCommunityRequests:
-      if (communityRequest.communityId == communityId):
+      if (communityRequest.communityId == communityId and communityRequest.state != int(RequestToJoinType.Loading)):
```

Comment:
I don't think we need the state check for `Loading` here.
As mentioned above, I don't think we need the loading state as a type at all.

Loading state can be handled entirely in the view. The idea would be to

1. Make request (set loading state to true)
2 When request finished, set loading state to false (now the request is pending)
3. If request fails due to an error, set loading state to false (ideally also show a notification that it failed due to some reason)